src/app/components/content/IsaacCoordinateQuestion.tsx

```diff
@@ -51,32 +50,26 @@ export const coordinateInputValidator = (input: string[][]) => {
 };
 
 const CoordinateInput = (props: CoordinateInputProps) => {
-    const {value, placeholderXValue, placeholderYValue, onChange, readonly, remove} = props;
-    return <span className="coordinate-input">
-        (
-        <Input
-            type="text"
-            className="force-print"
-            placeholder={placeholderXValue ?? "x"}
-            value={value.x ?? ""}
-            onChange={event => onChange({...value, x: event.target.value === "" ? undefined : event.target.value})}
-            readOnly={readonly}
-        />
-        <span className="coordinate-input-separator">,&nbsp;</span>
-        <Input
-            type="text"
-            className="force-print"
-            placeholder={placeholderYValue ?? "y"}
-            value={value.y ?? ""}
-            onChange={event => onChange({...value, y: event.target.value === "" ? undefined : event.target.value})}
-            readOnly={readonly}
-        />
-        )
-        {remove && <Button className="ms-3" size="sm" onClick={remove}>Delete</Button>}
+    const {value, placeholderValues, numberOfDimensions, onChange, readonly, remove} = props;
+    return <span className="coordinate-input">{[...Array(numberOfDimensions)].map((_, i) =>
+        <>
+            <Input
+                key={i}
+                type="text"
+                className="force-print"
+                placeholder={placeholderValues[i] ?? ""}
+                value={value.coordinates ? value.coordinates[i] : ""}
```

Comment:
`value.coordinates[i]` is undefined on first render. Since you're controlling the value of this, it's a controlled component, which [shouldn't have undefined values](https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable). 
```suggestion
                value={isDefined(value.coordinates?.[i]) ? value.coordinates[i] : ""}
```